src/pivot-table/components/cells/DimensionCell.tsx

```diff
@@ -99,9 +99,7 @@ const DimensionCell = ({
   const { select, isSelected, isActive, isLocked } = useSelectionsContext();
   const isNull = qType === NxDimCellType.NX_DIM_CELL_NULL;
   const selectionCellType = isLeftColumn ? NxSelectionCellType.NX_CELL_LEFT : NxSelectionCellType.NX_CELL_TOP;
-  const isCellLocked =
-    isLocked(selectionCellType, cell.y, colIndex) ||
-    layoutService.isDimensionLocked(selectionCellType, cell.y, colIndex);
+  const isCellLocked = isLocked(selectionCellType, cell.y, colIndex) || cell.isLockedByDimension;
```

Comment:
No because `isLocked` is checking if a dimension is locked becaus selections mode is active in another field. So it's not something that Engine is aware of. So it is not available in qDimensionInfo.

Reply:
To clarify. A cell should be locked for selections if any of the 2 following conditions are true:
1. Dimension is locked.
2. There are ongoing selections in another field.